Workspace API token documentation (#2018)
astronomer/docs: astro/workspace-api-tokens.md

3. Click **Edit** next to your API token.

4. Update the name, description, workspace role, or expiration date of your token, then click **Save Changes**.
   
5. Optional. To delete a Workspace API token, click **Delete API Token**, enter `Delete`, and then click **Yes, Continue**.

## Use a Workspace API token with the Astro CLI

To use a Workspace API token with the Astro CLI, you must make your API token available to the system running the Astro CLI using the `ASTRO_API_TOKEN` environment variable. 

For example, to automate Astro CLI Workspace commands on a Mac computer, run the following command to set a temporary value for the environment variable:

```sh
export ASTRO_API_TOKEN=<your-token>
```

After you set the variables, you can run `astro deployment` and `astro workspace` commands for your Workspace without authenticating yourself to Astronomer. Astronomer recommends storing `ASTRO_API_TOKEN` as a secret before using it to automate the Astro CLI for production workflows.

## Use an API token for CI/CD

You can use Workspace API tokens to automate various Workspace and Deployment management workflows in CI/CD. 

>  Regardless of your use case, you must make the following environment variable available to your CI/CD environment:

Author: ```suggestion
For all use cases, you must make the following environment variable available to your CI/CD environment:
```

> ```

Author: ```suggestion
```text
```

id: workspace-api-tokens
description: Create and manage Workspace API tokens to automate key Workspace actions like adding users and creating Deployments. 
---

Use Workspace API tokens to automate Workspace actions such as creating Deployments and managing users as part of your CI/CD pipelines. Using Workspace API tokens, you can automate:

- Creating and updating Deployments using a [Deployment file](manage-deployments-as-code.md)
- Adding batches of users to a Workspace in a CI/CD pipeline. See [Add a group of users to Astro using the Astro CLI](add-user.md#add-a-group-of-users-to-astro-using-the-astro-cli).
- Creating preview Deployments whenever you create a feature branch in your Astro project Git repository. 

## Create a Workspace API token

1. In the Cloud UI, open your Workspace.
   
2. Go to **Workspace Settings** > **Access Management** > **API Tokens**.
   
3. Click **API Token**.
   
4. Configure the new Workspace API token:

    - **Name**: The name for the API token.
    - **Description**: The Description for the API token.
    - **Workspace Role**: The role that the API token can assume. See [User permissions](user-permissions.md#workspace-roles).

>     - **Expiration**: The duration that the API token can be used before it expires.

Reviewer: This value should be in number of days.

5. Click **Create API token**. A confirmation screen showing the token appears.
   

> 6. Copy the token and store it in a safe place. You will not be able to retrieve this value from Astro again. 

Reviewer: Technically if they forget to store it in the safe place, they can rotate the token and get a fresh value

> ## Update or delete a Workspace API token

Reviewer: We should add a separate section for Rotating an Workspace API Token

If you delete a Workspace API token, make sure that no existing CI/CD pipelines are using it. Once deleted, an API token and secret cannot be recovered. If you unintentionally delete an API token, create a new one and update any CI/CD workflows that used the deleted API token.

1. In the Cloud UI, open your Workspace.
   
2. Go to **Workspace Settings** > **Access Management** > **API Tokens**.

3. Click **Edit** next to your API token.

> 4. Update the name, description, workspace role, or expiration date of your token, then click **Save Changes**.

Reviewer: You cannot update expiration date of the token.

   
5. Optional. To delete a Workspace API token, click **Delete API Token**, enter `Delete`, and then click **Yes, Continue**.

> ## Use a Workspace API token with the Astro CLI

Reviewer: This will be possible with cli version >= 1.12.0. We should highlight that

Author: We try to always assume latest for the Astro CLI, knowing that there haven't been breaking changes between versions for like a year. I think it's okay if someone tries and sees that they have to upgrade their CLI version

To use a Workspace API token with Astro CLI, specify the `ASTRO_API_TOKEN` environment variable in the system running the Astro CLI.  

For example, to automate Astro CLI Workspace commands on a Mac, run the following command to set a temporary value for the environment variable:

```sh
export ASTRO_API_TOKEN=<your-token>
```

After you set the variable, you can run `astro deployment` and `astro workspace` commands for your Workspace without authenticating yourself to Astronomer. Astronomer recommends storing `ASTRO_API_TOKEN` as a secret before using it to automate the Astro CLI for production workflows.

:::info

> If you have both `ASTRO_API_TOKEN` and `ASTRONOMER_KEY_ID`/`ASTRONOMER_KEY_SECRET` set in an environment, your Astro Workspace token takes precedence and is used for all Deployment actions. 

Reviewer: ```suggestion
If you have both `ASTRO_API_TOKEN` and `ASTRONOMER_KEY_ID`/`ASTRONOMER_KEY_SECRET` set in an environment, your Astro Workspace token takes precedence and is used for all Deployment actions in that workspace. 
```

- Creating preview Deployments whenever you create a feature branch in your Astro project Git repository. 
- Performing Deployment-level actions on any Deployment in a Workspace, such as deploying code or making calls to the Airflow rest API. Workspace API tokens can complete the same actions as Deployment API keys for any Deployment in the Workspace. 

## Create a Workspace API token

1. In the Cloud UI, open your Workspace.
   
2. Go to **Workspace Settings** > **Access Management** > **API Tokens**.
   
3. Click **API Token**.
   
4. Configure the new Workspace API token:

    - **Name**: The name for the API token.
    - **Description**: Optional. The Description for the API token.
    - **Workspace Role**: The role that the API token can assume. See [User permissions](user-permissions.md#workspace-roles).
    - **Expiration**: The number of days that the API token can be used before it expires.

5. Click **Create API token**. A confirmation screen showing the token appears.
   
6. Copy the token and store it in a safe place. You will not be able to retrieve this value from Astro again. 

## Update or delete a Workspace API token

> If you delete a Workspace API token, make sure that no existing CI/CD pipelines are using it. Once deleted, an API token cannot be recovered. If you unintentionally delete an API token, create a new one and update any CI/CD workflows that used the deleted API token.

Reviewer 2: This is a grammar/style nit I have - English speakers often use the term `once` when we mean `after`. It can be a little confusing because "Once event occurs" can actually refer to an event that happens many, many times. "Once" can be a little confusing in that context in contrast to "After the event occurs" doesn't have that very subtle ambiguity.

> If you delete a Workspace API token, make sure that no existing CI/CD pipelines are using it. Once deleted, an API token cannot be recovered. If you unintentionally delete an API token, create a new one and update any CI/CD workflows that used the deleted API token.

Author: ```suggestion
If you delete a Workspace API token, make sure that no existing CI/CD pipelines are using it. After it's deleted, an API token cannot be recovered. If you unintentionally delete an API token, create a new one and update any CI/CD workflows that used the deleted API token.
```

1. In the Cloud UI, open your Workspace.
   
2. Go to **Workspace Settings** > **Access Management** > **API Tokens**.

3. Click **Edit** next to your API token.

4. Update the name, description, or Workspace role of your token, then click **Save Changes**.
   
5. Optional. To delete a Workspace API token, click **Delete API Token**, enter `Delete`, and then click **Yes, Continue**.

## Rotate a Workspace API token

Rotating a Workspace API token lets you renew a token without needing to reconfigure its name, description, and permissions. You can also rotate a token if you lose your current token value and need it for additional workflows. 

> When you rotate a Workspace API token, you receive a new valid token that can be used in your existing workflows. The previous token value becomes invalid and any workflows using the old value stop working. 

Reviewer 2: ```suggestion
When you rotate a Workspace API token, you receive a new valid token from Astro that can be used in your existing workflows. The previous token value becomes invalid and any workflows using those previous values stop working. 
```
